peer_set.py

```diff
@@ -0,0 +1,36 @@
+from time import time
+from typing import Iterable, Optional
+
+from pynanocoin import Peer
+
+
+class peer_set(set):
```

Comment:
What about thread protection. Don;t we need a mutex?

Reply:
It seems that peer_set is not thread protected and expects the caller to do thread protection.